libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -897,6 +899,123 @@ std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::resizeDynamicByteArrayFunction(ArrayType const& _type)
+{
+	string functionName = "resize_array_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(array, newLen) {
+				if gt(newLen, <maxArrayLength>) {
+					<panic>()
+				}
+
+				let oldLen := <fetchLength>(array)
+				let offset := mod(newLen, 32)
+				let oldSlotCount := div(add(oldLen, 31), 32)
+				let newSlotCount := div(add(newLen, 31), 32)
+
+				if gt(newLen, oldLen) {
+					<increaseSize>(array, oldLen, newLen)
+				}
+
+				// Size was reduced, clear end of array
+				if lt(newLen, oldLen) {
+					switch lt(newSlotCount, 2)
```

Comment:
Don't you need three cases?
 - going from larger than 31 to larger than 31
 - going from larger than 31 to smaller than 31
 - going from smaller than 31 to smaller than 31

Reply:
First case is handled here, second and third in `<decreaseSizeToPacked>` call